packages/react/src/components/DatePicker/DatePicker.tsx

```diff
@@ -444,31 +444,38 @@ const DatePicker = React.forwardRef(function DatePicker(
   const lastStartValue = useRef('');
 
   // fix datepicker deleting the selectedDate when the calendar closes
-  const onCalendarClose = (selectedDates, dateStr) => {
-    endInputField?.current?.focus();
-    calendarRef?.current?.calendarContainer?.classList.remove('open');
-    setTimeout(() => {
-      if (
-        lastStartValue.current &&
-        selectedDates[0] &&
-        !startInputField.current.value
-      ) {
-        startInputField.current.value = lastStartValue.current;
-        calendarRef.current.setDate(
-          [startInputField.current.value, endInputField?.current?.value],
-          true,
-          calendarRef.current.config.dateFormat
-        );
-      }
-      if (onClose) {
-        onClose(
-          calendarRef.current.selectedDates,
-          dateStr,
-          calendarRef.current
-        );
+  const onCalendarClose = useCallback(
```

Comment:
`savedOnClose` is already wrapping `onCalendarClose` in a useCallback. Could the new useCallback be removed?